### DISCUSSION TITLE
Add maxPrecision to ZodNumber

### DISCUSSION THREAD

**Discussion by rubber-duck-software** [colinhacks/zod#1952] [Q&A] [answered]
This would allow for quickly defining currency types where decimal places are allowed, but `1.001` is not valid.

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const schema = z.number().refine( n => {
    return n.toString().split( '.' )[ 1 ].length <= 2
}, { message: 'Max precision is 2 decimal places' } )

console.log( schema.safeParse( 1.01 ).success ) // true
console.log( schema.safeParse( 1.001 ).success ) // false
```

**Comment by rubber-duck-software** [comment]
Yes! As I have been thinking about this, it would not only be helpful for currency types, but also relational databases often use fixed decimal representation which javascript floating point representation can overflow.

**Comment by rubengmurray** [comment]
I support this, and I would expect it to error if > the provided value (as is the case in the written `.refine(...)` above)

I'm currently making the switch to `zod` having historically used `joi`. `joi` has the method `precision`, but in `14.3.1` at least it doesn't throw (it just rounds) which I don't particularly like (and proved to be a gotcha whilst implementing).
